packages/cli/lib/utils.js

```diff
@@ -295,3 +297,43 @@ exports.validateStringObject = function(type) {
     return true;
   };
 };
+
+/**
+ * Validate property name
+ * @param {String} name The user input
+ * @returns {String|Boolean}
+ */
+exports.checkPropertyName = function(name) {
+  var result = exports.validateRequiredName(name);
+  if (result !== true) return result;
+  if (RESERVED_PROPERTY_NAMES.includes(name)) {
+    return `${name} is a reserved keyword. Please use another name`;
+  }
+  return true;
+};
+
+/**
+ * Validate required name for properties, data sources, or connectors
+ * Empty name could not pass
+ * @param {String} name The user input
+ * @returns {String|Boolean}
+ */
+exports.validateRequiredName = function(name) {
+  if (!name) {
+    return 'Name is required';
+  }
+  return validateValue(name, /[\/@\s\+%:\.]/);
+};
+
+function validateValue(name, unallowedCharacters) {
+  if (!unallowedCharacters) {
+    unallowedCharacters = /[\/@\s\+%:\.]/;
+  }
+  if (name.match(unallowedCharacters)) {
+    return `Name cannot contain special characters ${unallowedCharacters} ${name}`;
```

Comment:
missing a colon between the two variables

Reply:
Also, consider using parts of this: https://github.com/strongloop/loopback-next/pull/1487/commits/d6f1a64d817a79f9b74f504374b484da99e57faf#diff-690866a08191a550353b1a1929fc6080R38